pkg/sql2pgroll/create_index.go

```diff
@@ -18,12 +19,62 @@ func convertCreateIndexStmt(stmt *pgq.IndexStmt) (migrations.Operations, error)
 
 	// Get the qualified table name
 	tableName := getQualifiedRelationName(stmt.GetRelation())
-	var columns []string
 
-	// Get the columns on which the index is defined
+	// Get the columns and their settings on which the index is defined
+	columns := make(map[string]migrations.IndexElemSettings, len(stmt.GetIndexParams()))
 	for _, param := range stmt.GetIndexParams() {
 		if colName := param.GetIndexElem().GetName(); colName != "" {
-			columns = append(columns, colName)
+			var indexElemSettings migrations.IndexElemSettings
+			collate, err := pgq.DeparseAnyName(param.GetIndexElem().GetCollation())
+			if err != nil {
+				return nil, nil
+			}
+			indexElemSettings.Collate = collate
+			opclassName, err := pgq.DeparseAnyName(param.GetIndexElem().GetOpclass())
+			if err != nil {
+				return nil, nil
+			}
+			if opclassName != "" {
+				opclassOpts := make(map[string]any, 0)
+				opts, err := pgq.DeparseRelOptions(param.GetIndexElem().GetOpclassopts())
+				if err != nil {
+					return nil, nil
+				}
+				for _, opt := range strings.Split(opts[1:len(opts)-1], ",") {
+					optKV := strings.SplitN(opt, "=", 2)
+					if len(optKV) != 2 {
+						continue
+					}
+					opclassOpts[optKV[0]] = optKV[1]
+				}
```

Comment:
Do we need to store these opclass options in a map?

Could the format be:

```json
{
  "name": "54_create_index_with_opclass",
  "operations": [
    {
      "create_index": {
        "name": "idx_fruits_custom_opclass",
        "table": "fruits",
        "columns": {
          "id": {
            "opclass": {
              "name": "int8_ops"
               "params": "foo = bar"
            }
          }
        }
      }
    }
  ]
}
```

Then we wouldn't need this manual deparsing. I don't think we use the extra structure of having the opclass options as a map.